README.md

```diff
@@ -10,213 +10,320 @@
+Amazon Kinesis Video Streams Producer SDK for C/C++ makes it easy to build an on-device application that securely connects to a video stream, and reliably publishes video and other media data to Kinesis Video Streams (KVS). It takes care of all the underlying tasks required to package the frames and fragments generated by the device's media pipeline. The SDK also handles stream creation, token rotation for secure and uninterrupted streaming, processing acknowledgements returned by Kinesis Video Streams, and other tasks. The SDK includes a GStreamer plugin to stream directly to a custom GStreamer element kvssink, and includes a JNI for use with Java applications and the [KVS Java Producer SDK](https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-java).
+
+<br>
+
+For more info on KVS please see [AWS KVS Documentation](https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/what-is-kinesis-video.html) and [Amazon Kinesis Video Streams FAQs](https://aws.amazon.com/kinesis/video-streams/faqs/).
+
+<br>
+
+## Quick Start
+### Required Tools
+The following packages are required to build the SDK libraries. Using a package manager such as _Homebrew_ (Mac), _APT_ (Linux), and _Chocolatey_ (Windows) is the prefered method of installation.
+* C++ Compiler (GNU or Clang recommended)
+* `git`
+* `CMake`
+* `pkg-config` for _Mac_/_Linux_, `pkgconfiglite` for _Windows_
+* `m4`
+* _Windows_ only: `nasm` and `strawberryperl`
+
+<br>
+
+### Install GStreamer
+If building the samples or the kvssink GStreamer plugin, GStreamer libraries are required.
 
+_Mac_
 ```bash
-git clone https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp.git
+brew install gstreamer gst-plugins-base gst-plugins-good gst-plugins-bad gst-plugins-ugly gst-libav
+```
+_Linux_
+```bash
+sudo apt install libgstreamer1.0-dev libgstreamer-plugins-base1.0-dev gstreamer1.0-plugins-base-apps gstreamer1.0-plugins-bad gstreamer1.0-plugins-good gstreamer1.0-plugins-ugly gstreamer1.0-tools
+```
+_Windows_
+```bat
+choco install gstreamer --version=1.22.8
+choco install gstreamer-devel --version=1.22.8
+```
+#### Verify GStreamer Installation
+Run the following command to display the GStreamer version to confirm the installation was successful:
+```bash
+gst-launch-1.0 --gst-version
 ```
 
-Note: You will also need to install `pkg-config`, `CMake`, `m4` and a build enviroment. If you are building the GStreamer plugin you will also need GStreamer and GStreamer (Development Libraries).
-
-Refer to the [FAQ](#FAQ) for platform specific instructions.
+<br>
 
-### Configure
+### Download
+```bash
+git clone https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp.git
+```
 
-Prepare a build directory in the newly checked out repository:
+<br>
 
+### Build
+#### Prepare a build directory in the newly checked out repository:
 ```bash
 mkdir -p amazon-kinesis-video-streams-producer-sdk-cpp/build
 cd amazon-kinesis-video-streams-producer-sdk-cpp/build
```

Comment:
Don't we need to clone it in the `C:/` for windows due to file path length restrictions?

Also can we adjust this to clone only the master branch?

Reply:
If you could add a note for that one, that would be great.

Reply:
Not needed when building Windows locally, we move it in github actions because it has it nested in `amazon-kinesis-video-streams-producer-sdk-cpp/amazon-kinesis-video-streams-producer-sdk-cpp` which is not going to be the case when cloning locally.